src/libraries/System.Text.RegularExpressions/gen/UpgradeToRegexGeneratorAnalyzer.cs

```diff
@@ -39,9 +39,9 @@ public override void Initialize(AnalysisContext context)
             context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
             context.EnableConcurrentExecution();
 
-            context.RegisterCompilationStartAction(async compilationContext =>
+            context.RegisterCompilationStartAction(context =>
```

Comment:
Shadowing the outer `context` is intentional by the rename. It prevents accidental usage of the outer `context` when the inner `context` must be use (e.g, accessing cancellation token must be from the most inner context)

Reply:
The idea here makes sense. I'm not sure how clear this would be in terms of code readability. At the very least I would suggest to add a comment on a line above explaining why we are using the same variable name. Adding @stephentoub for his thoughts around variable shadowing and whether or not we use this elsewhere in the repo.

Reply:
fwiw, this was suggested by @sharwell in some PR in dotnet/roslyn-analyzers in past.

Reply:
Yeah, the idea makes sense, I think it is a good suggestion, my only concern is readability, but as I said earlier I think that could be addressed by adding a comment to mention our intent.

Reply:
shadowing is fine. we do use it elsewhere.